src/Areas/Server/Commands/ToolLoading/CommandFactoryToolLoader.cs

```diff
@@ -4,67 +4,65 @@
 using System.Diagnostics;
 using System.Reflection;
 using System.Text.Json.Nodes;
+using AzureMcp.Areas.Server.Options;
 using AzureMcp.Commands;
 using AzureMcp.Services.Telemetry;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using ModelContextProtocol;
 using ModelContextProtocol.Protocol;
 using static AzureMcp.Services.Telemetry.TelemetryConstants;
 
-namespace AzureMcp.Areas.Server.Commands;
-
-public class ToolOperations
+namespace AzureMcp.Areas.Server.Commands.ToolLoading;
+
+/// <summary>
+/// A tool loader that creates MCP tools from the registered command factory.
+/// Exposes AzureMcp commands as MCP tools that can be invoked through the MCP protocol.
+/// </summary>
+public sealed class CommandFactoryToolLoader(
+    IServiceProvider serviceProvider,
+    CommandFactory commandFactory,
+    IOptions<ServiceStartOptions> options,
+    ITelemetryService telemetry,
+    ILogger<CommandFactoryToolLoader> logger) : IToolLoader
 {
-    private readonly IServiceProvider _serviceProvider;
-    private readonly CommandFactory _commandFactory;
-    private readonly ITelemetryService _telemetry;
-    private IReadOnlyDictionary<string, IBaseCommand> _toolCommands;
-    private readonly ILogger<ToolOperations> _logger;
-    private string[]? _commandGroup = null;
-
-    public ToolOperations(IServiceProvider serviceProvider, CommandFactory commandFactory, ITelemetryService telemetry, ILogger<ToolOperations> logger)
+    private readonly IServiceProvider _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
```

Comment:
There is parameter null checking so you don't have to do ArgumentNullException check any longer.

Reply:
There is a lot of other services still throwing Arg null exceptions.  Lets do a full pass after this PR if we want to apply a consistent pattern.